rnn_tests/test_node_to_binary.cxx

```diff
@@ -0,0 +1,108 @@
+#include <chrono>
+#include <fstream>
+using std::getline;
+using std::ifstream;
+
+#include <random>
+using std::minstd_rand0;
+using std::uniform_real_distribution;
+
+#include <string>
+using std::string;
+
+#include <vector>
+using std::vector;
+
+#include "common/arguments.hxx"
+#include "common/log.hxx"
+#include "gradient_test.hxx"
+#include "rnn/generate_nn.hxx"
+#include "rnn/rnn_edge.hxx"
+#include "rnn/rnn_genome.hxx"
+#include "rnn/rnn_node.hxx"
+#include "rnn/rnn_node_interface.hxx"
+#include "rnn/sin_node.hxx"
+#include "time_series/time_series.hxx"
+#include "weights/weight_rules.hxx"
+
+int main(int argc, char** argv) {
+    vector<string> arguments = vector<string>(argv, argv + argc);
+
+    Log::initialize(arguments);
+    Log::set_id("main");
+
+    initialize_generator();
+
+    RNN_Genome* genome_original = nullptr;
+
+    vector<vector<double> > inputs;
+    vector<vector<double> > outputs;
+
+    int input_length = 10;
+    string hidden_node_type;
+    get_argument(arguments, "--hidden_node_type", true, hidden_node_type);
+
+    WeightRules* weight_rules = new WeightRules();
+
+    int32_t max_recurrent_depth = 3;
+    Log::info("testing with max recurrent depth: %d\n", max_recurrent_depth);
+
+    inputs.resize(1);
+    outputs.resize(1);
+
+    generate_random_vector(input_length, inputs[0]);
+    generate_random_vector(input_length, outputs[0]);
+
+    vector<string> inputs1{"input 1"};
+    vector<string> outputs1{"output 1"};
+
+    if (hidden_node_type.compare("sin") == 0) {
+        Log::info("TESTING SIN!!!\n");
+        genome_original = create_sin(inputs1, 1, 2, outputs1, max_recurrent_depth, weight_rules);
+    }
+    string path = "/Users/jaredmurphy/exact/scripts/node_tests/genome_original.bin";
+    genome_original->write_to_file(path);
+    RNN_Genome* genome_from_file = new RNN_Genome(path);
+
+    int32_t num_weights = genome_original->get_number_weights();
+    vector<double> weights;
+    generate_random_vector(num_weights, weights);
+
+    RNN* rnn_original = genome_original->get_rnn();
+    RNN* rnn_from_file = genome_from_file->get_rnn();
+
+    double analytic_mse_original, analytic_mse_file;
+    vector<double> analytic_gradient_original, analytic_gradient_file;
+
+    rnn_original->get_analytic_gradient(
+        weights, inputs, outputs, analytic_mse_original, analytic_gradient_original, false, true, 0.0
+    );
+    rnn_from_file->get_analytic_gradient(
+        weights, inputs, outputs, analytic_mse_file, analytic_gradient_file, false, true, 0.0
+    );
+
```

Comment:
also run the get_backprop_gradient on both and check if those are the same (maybe with different inputs/outputs than for the analytic gradient)